sdk/storage/azure-storage-file-share/azure/storage/fileshare/aio/_file_client_async.py

```diff
@@ -432,6 +432,19 @@ async def create_file(
             NFS only. The owning group of the file.
         :keyword str file_mode:
             NFS only. The file mode of the file.
+        :keyword file_property_semantics:
+            SMB only. Specifies permissions to be configured. Default value is None.
+            If not specified or None is passed, New will be the default. Possible values are:
+
+                New - forcefully add the ARCHIVE attribute flag and alter the permissions specified in
+                x-ms-file-permission to inherit missing permissions from the parent.
+
+                Restore - apply changes without further modification.
+
+        :paramtype file_property_semantics: Literal["New", "Restore"]
+        :keyword data: Initial data to upload. The limit is 4MB.
+        :paramtype data: IO[bytes] or bytes
+        :keyword int length: Specifies the number of bytes being uploaded in the initial data.
```

Comment:
It might be worth clarifying the behaviour here - is this requried if I'm passing in `data`? Will it only be required if length cannot be easily determined from `data`? Will it be used even if it conflicts with `len(data)`?